src/main/java/com/fasterxml/jackson/databind/DeserializationContext.java

```diff
@@ -469,10 +469,12 @@ public final Object findInjectableValue(Object valueId,
         throws JsonMappingException
     {
         if (_injectableValues == null) {
-            // `optional` comes from property annotation (if any); has precedence
-            // over global setting.
-            if (Boolean.TRUE.equals(optional)
-                    || (optional == null && !isEnabled(DeserializationFeature.FAIL_ON_UNKNOWN_INJECT_VALUE))) {
+            // `useInput` and `optional` come from property annotation (if any);
+            // they have precedence over global setting.
+            if (Boolean.TRUE.equals(useInput)
+                    || Boolean.TRUE.equals(optional)
+                    || ((useInput == null || optional == null)
+                            && !isEnabled(DeserializationFeature.FAIL_ON_UNKNOWN_INJECT_VALUE))) {
                 return JacksonInject.Value.empty();
```

Comment:
Hmmmh. This is wrong -- should be `null`, its Injectable value, NOT injection definition.

Reply:
Changed in 2.20, looks like; quietly returning new marker value. Should have caught this problem back then; I think I'll this now for 2.21.